### ISSUE TITLE
Uncaught TypeError: Cannot read property 'remove' of undefined

### ISSUE THREAD

**Issue by KyleOh** [rgbkrk/atom-script#1224]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Mac OS X 10.12.3
**Thrown From**: [script](https://github.com/rgbkrk/atom-script) package, v3.14.0

### Stack Trace

Uncaught TypeError: Cannot read property 'remove' of undefined

```
At /Users/JaeWook/.atom/packages/script/node_modules/atom-message-panel/lib/MessagePanelView.js:189

TypeError: Cannot read property 'remove' of undefined
    at ScriptView.MessagePanelView.remove (/Users/JaeWook/.atom/packages/script/node_modules/atom-message-panel/lib/MessagePanelView.js:189:30)
    at ScriptView.detach (/Users/JaeWook/.atom/packages/script/node_modules/jquery/dist/jquery.js:5357:15)
    at ScriptView.removePanel (/Users/JaeWook/.atom/packages/script/lib/script-view.js:89:10)
    at Object.closeScriptViewAndStopRunner (/Users/JaeWook/.atom/packages/script/lib/script.js:161:21)
    at atom-workspace.scriptCloseView (/Users/JaeWook/.atom/packages/script/lib/script.js:94:39)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:259:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:59)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at ScriptView.close (/Users/JaeWook/.atom/packages/script/lib/script-view.js:99:19)
    at HTMLDivElement.<anonymous> (/Users/JaeWook/.atom/packages/script/node_modules/space-pen/lib/space-pen.js:220:36)
    at HTMLDivElement.dispatch (/Users/JaeWook/.atom/packages/script/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLDivElement.elemData.handle (/Users/JaeWook/.atom/packages/script/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -2:14.5.0 editor:indent (input.hidden-input)
     -2:14 core:backspace (input.hidden-input)
     -2:13.6.0 core:move-right (input.hidden-input)
     -2:13.3.0 editor:newline (input.hidden-input)
  9x -2:08.2.0 core:backspace (input.hidden-input)
  2x -2:03.1.0 core:move-right (input.hidden-input)
     -1:55.9.0 fuzzy-finder:toggle-git-status-finder (input.hidden-input)
  2x -1:54.7.0 script:run (input.hidden-input)
  2x -1:06.2.0 script:close-view (atom-workspace.workspace.scrollbars-visible-when-scrolling.use-animations.theme-atom-material-syntax.theme-atom-material-ui)
     -0:47.2.0 core:undo (input.hidden-input)
     -0:29.2.0 core:paste (div.package-detail.panels-item)
     -0:28.7.0 core:copy (div.package-detail.panels-item)
  2x -0:28 script:kill-process (div.package-detail.panels-item)
     -0:25.2.0 script:close-view (atom-workspace.workspace.scrollbars-visible-when-scrolling.use-animations.theme-atom-material-syntax.theme-atom-material-ui)
  4x -0:07.3.0 core:copy (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -0:00.0 script:close-view (atom-workspace.workspace.scrollbars-visible-when-scrolling.use-animations.theme-atom-material-syntax.theme-atom-material-ui)
```

### Config

```json
{
  "core": {
    "openEmptyEditorOnStart": false,
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v1.0.2 (active)
atom-material-ui, v1.3.9 (active)
emmet, v2.4.3 (active)
language-jade, v0.7.2 (active)
markdown-preview-enhanced, v0.9.3 (active)
script, v3.14.0 (active)
tabs-to-spaces, v1.0.3 (active)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (inactive)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by bmweiner** [comment]
This error is thrown when I hit the escape key

**Comment by martianbaby** [comment]
I only installed Atom along with a few packages today but I have this problem too.

I'm unable to close the affected tab using CMD+W after I use CMD+I. Manually clicking the X button in the tab bar works though. It's also impossible to close the output pane as the error occurs when clicking 'X' icon in the pane but collapsing/minimising the pane is fine.

[Enter steps to reproduce below:]

1. Run script with CMD+I shortcut.
2. Output displayed in the output pane.
3. Click 'X' icon in the output pane or use CMD+W to close tab.

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Mac OS X 10.12.2
**Thrown From**: [script](https://github.com/rgbkrk/atom-script) package, v3.14.0


### Stack Trace

Uncaught TypeError: Cannot read property 'remove' of undefined

```
At /Users/DeadRin/.atom/packages/script/node_modules/atom-message-panel/lib/MessagePanelView.js:189

TypeError: Cannot read property 'remove' of undefined
    at ScriptView.MessagePanelView.remove (/Users/DeadRin/.atom/packages/script/node_modules/atom-message-panel/lib/MessagePanelView.js:189:30)
    at ScriptView.detach (/Users/DeadRin/.atom/packages/script/node_modules/jquery/dist/jquery.js:5357:15)
    at ScriptView.removePanel (/Users/DeadRin/.atom/packages/script/lib/script-view.js:89:10)
    at Object.closeScriptViewAndStopRunner (/Users/DeadRin/.atom/packages/script/lib/script.js:161:21)
    at atom-workspace.scriptCloseView (/Users/DeadRin/.atom/packages/script/lib/script.js:94:39)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:259:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:59)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at ScriptView.close (/Users/DeadRin/.atom/packages/script/lib/script-view.js:99:19)
    at HTMLDivElement.<anonymous> (/Users/DeadRin/.atom/packages/script/node_modules/space-pen/lib/space-pen.js:220:36)
    at HTMLDivElement.dispatch (/Users/DeadRin/.atom/packages/script/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLDivElement.elemData.handle (/Users/DeadRin/.atom/packages/script/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -9:51.6.0 script:run (input.hidden-input)
  2x -9:50.1.0 core:close (input.hidden-input)
     -7:59.5.0 settings-view:open (input.hidden-input)
  2x -7:57.8.0 core:close (input.hidden-input)
     -7:32.4.0 core:backspace (input.hidden-input)
     -6:48.2.0 core:close (input.hidden-input)
  6x -4:25.8.0 script:close-view (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-solarized-light-syntax.theme-one-light-ui)
  5x -3:57.3.0 script:kill-process (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -3:49.4.0 core:copy (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -3:44.2.0 script:run (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -3:42.6.0 script:kill-process (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -3:36.8.0 script:close-view (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-solarized-light-syntax.theme-one-light-ui)
     -3:29.9.0 script:run (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -3:27.5.0 core:close (div.am-panel.tool-panel.panel-bottom.native-key-bindings.script-view)
     -0:00.0 script:close-view (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-solarized-light-syntax.theme-one-light-ui)
```

### Config

```json
{
  "core": {
    "packagesWithKeymapsDisabled": [
      "highlight-line"
    ],
    "telemetryConsent": "no",
    "themes": [
      "one-light-ui",
      "solarized-light-syntax"
    ],
    "useCustomTitleBar": true
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.17 (inactive)
atom-bracket-highlight, v0.3.1 (active)
auto-detect-indentation, v1.3.0 (active)
auto-semicolon, v1.1.0 (active)
autocomplete-python, v1.8.27 (active)
color-picker, v2.2.4 (active)
highlight-line, v0.12.0 (active)
highlight-selected, v0.12.0 (active)
indent-guide-improved, v1.4.12 (active)
linter, v1.11.21 (active)
linter-pylint, v1.2.2 (active)
minimap, v4.26.7 (active)
open-recent, v5.0.0 (active)
pigments, v0.39.0 (active)
script, v3.14.0 (active)
Sublime-Style-Column-Selection, v1.7.2 (active)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (inactive)
one-light-ui, v1.8.2 (active)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (active)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by lgeiger** [comment]
This exception was fixed in #1223 though we haven't shipped a patch release yet.
I can ship a release later today.

**Comment by lgeiger** [comment]
Published as `v3.14.1` 🚀
